### PR TITLE
[Feature] VLanId property of VDPortGroup DSC Resource

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- Added **VLanId** property of **VDPortGroup DSC Resource**.
- Added Unit Tests for **VLanId** property of **VDPortGroup DSC Resource**.
- Added Integration Tests for **VLanId** property of **VDPortGroup DSC Resource**.
- Added Documentation and example Configuration for **VLanId** property of **VDPortGroup DSC Resource**.

### CODE REVIEW

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 1]
```diff
@@ -287,10 +378,18 @@ class VDPortGroup : BaseDSC {
     #>
     [void] RemoveDistributedPortGroup($distributedPortGroup) {
         try {
-            $distributedPortGroup | Remove-VDPortGroup -Server $this.Connection -Confirm:$false -ErrorAction Stop
+            Write-VerboseLog -Message $this.RemoveVDPortGroupMessage -Arguments @($distributedPortGroup.Name, $distributedPortGroup.VDSwitch.Name)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:381]: **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 2]
```diff
@@ -273,10 +334,40 @@ class VDPortGroup : BaseDSC {
...
+
+        if ($this.VLanId -eq 0) {
+            $setVDVlanConfigurationParams.DisableVlan = $true
+        }
+        else {
+            $setVDVlanConfigurationParams.VlanId = $this.VLanId
+        }
+
+        try {
+            Write-VerboseLog -Message $this.ModifyVDPortGroupVlanConfigurationMessage -Arguments @($distributedPortGroup.Name, $this.VLanId)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:366]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 3]
```diff
@@ -273,10 +334,40 @@ class VDPortGroup : BaseDSC {
         $distributedPortGroupParams = $this.GetDistributedPortGroupParams()
 
         try {
+            Write-VerboseLog -Message $this.ModifyVDPortGroupMessage -Arguments @($distributedPortGroup.Name)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:337]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 4]
```diff
@@ -251,16 +310,18 @@ class VDPortGroup : BaseDSC {
...
-        ReferencePortGroup is parameter only for the New-VDPortgroup cmdlet
-        and is not used for the Set-VDPortgroup cmdlet.
+        ReferencePortGroup and VLanId are parameters only for the New-VDPortgroup cmdlet
+        and are not used for the Set-VDPortgroup cmdlet.
         #>
         if (![string]::IsNullOrEmpty($this.ReferenceVDPortGroupName)) { $distributedPortGroupParams.ReferencePortgroup = $this.ReferenceVDPortGroupName }
+        if ($null -ne $this.VLanId) { $distributedPortGroupParams.VlanId = $this.VLanId }
 
         try {
+            Write-VerboseLog -Message $this.CreateVDPortGroupMessage -Arguments @($this.Name, $distributedSwitch.Name)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:320]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 5]
```diff
@@ -148,6 +181,7 @@ class VDPortGroup : BaseDSC {
         }
         finally {
             $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.GetMethodEndMessage -Arguments @($this.DscResourceName)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:184]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 6]
```diff
@@ -129,11 +160,13 @@ class VDPortGroup : BaseDSC {
         }
         finally {
             $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.TestMethodEndMessage -Arguments @($this.DscResourceName)
         }
     }
 
     [VDPortGroup] Get() {
         try {
+            Write-VerboseLog -Message $this.GetMethodStartMessage -Arguments @($this.DscResourceName)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:169]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 7]
```diff
@@ -129,11 +160,13 @@ class VDPortGroup : BaseDSC {
         }
         finally {
             $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.TestMethodEndMessage -Arguments @($this.DscResourceName)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:163]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 8]
```diff
@@ -99,11 +128,13 @@ class VDPortGroup : BaseDSC {
         }
         finally {
             $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.SetMethodEndMessage -Arguments @($this.DscResourceName)
         }
     }
 
     [bool] Test() {
         try {
+            Write-VerboseLog -Message $this.TestMethodStartMessage -Arguments @($this.DscResourceName)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:137]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 9]
```diff
@@ -99,11 +128,13 @@ class VDPortGroup : BaseDSC {
         }
         finally {
             $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.SetMethodEndMessage -Arguments @($this.DscResourceName)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:131]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 10]
```diff
@@ -69,14 +69,37 @@ class VDPortGroup : BaseDSC {
...
+
+    hidden [string] $CouldNotRetrieveVDSwitchMessage = "Could not retrieve distributed switch {0}. For more information: {1}"
+    hidden [string] $CouldNotCreateVDPortGroupMessage = "Could not create distributed port group {0} on distributed switch {1}. For more information: {2}"
+    hidden [string] $CouldNotModifyVDPortGroupMessage = "Could not modify distributed port group {0}. For more information: {1}"
+    hidden [string] $CouldNotModifyVDPortGroupVlanConfigurationMessage = "Could not modify the VLAN ID of distributed port group {0} to {1}. For more information: {2}"
+    hidden [string] $CouldNotRemoveVDPortGroupMessage = "Could not remove distributed port group {0} from distributed switch {1}. For more information: {2}"
+
     [void] Set() {
         try {
+            Write-VerboseLog -Message $this.SetMethodStartMessage -Arguments @($this.DscResourceName)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:102]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

File: Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1 [hunk 11]
```diff
@@ -160,23 +194,30 @@ class VDPortGroup : BaseDSC {
...
             if ($this.Ensure -eq [Ensure]::Absent) {
-                return Get-VDSwitch -Server $this.Connection -Name $this.VdsName -ErrorAction SilentlyContinue
+                $getVDSwitchParams.ErrorAction = 'SilentlyContinue'
+                return Get-VDSwitch @getVDSwitchParams
             }
             else {
                 try {
-                    $distributedSwitch = Get-VDSwitch -Server $this.Connection -Name $this.VdsName -ErrorAction Stop
-                    return $distributedSwitch
+                    Write-VerboseLog -Message $this.RetrieveVDSwitchMessage -Arguments @($this.VdsName)
```
Comment (KristiyanGK) [inline Source/VMware.vSphereDSC/DSCResources/VDPortGroup.ps1:215]: Here as well **Write-VerboseLog** will need to be replaced by **WriteLogUtil** after the other PR gets approved.

**Review by KristiyanGK — APPROVED** [review]
Looks good!

**Review by dmilov — APPROVED** [review]

**Pull request merged**